tests/Fixer/FunctionNotation/PhpdocToReturnTypeFixerTest.php

```diff
@@ -0,0 +1,231 @@
+                70100,
+            ],
+            'invalid void return on ^7.1' => [
+                '<?php /** @return null|void */ function my_foo() {}',
+            ],
+            'iterable return on ^7.1' => [
+                '<?php /** @return iterable */ function my_foo(): iterable {}',
+                '<?php /** @return iterable */ function my_foo() {}',
+                70100,
+            ],
+            'object return on ^7.2' => [
+                '<?php /** @return object */ function my_foo(): object {}',
+                '<?php /** @return object */ function my_foo() {}',
+                70200,
+            ],
+            'no scalar types by default' => [
+                '<?php /** @return int */ function my_foo() {}',
+            ],
+            'fix scalar types when configured' => [
+                '<?php /** @return int */ function my_foo(): int {}',
+                '<?php /** @return int */ function my_foo() {}',
+                null,
+                ['scalar_types' => true],
+            ],
+            'array native type' => [
+                '<?php /** @return array */ function my_foo(): array {}',
+                '<?php /** @return array */ function my_foo() {}',
+            ],
+            'callable type' => [
+                '<?php /** @return callable */ function my_foo(): callable {}',
+                '<?php /** @return callable */ function my_foo() {}',
+            ],
+            'self accessor' => [
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo() {}
+                    }
+                ',
+            ],
+            'report static as self' => [
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo() {}
+                    }
+                ',
+            ],
+            'skip resource special type' => [
+                '<?php /** @return resource */ function my_foo() {}',
+            ],
+            'skip mixed special type' => [
+                '<?php /** @return mixed */ function my_foo() {}',
+            ],
+            'null alone cannot be a return type' => [
+                '<?php /** @return null */ function my_foo() {}',
+            ],
+            'skip mixed types' => [
+                '<?php /** @return Foo|Bar */ function my_foo() {}',
+            ],
+            'nullable type' => [
+                '<?php /** @return null|Bar */ function my_foo(): ?Bar {}',
+                '<?php /** @return null|Bar */ function my_foo() {}',
+                70100,
```

Comment:
super trivial, but would be nice to have test-case showing behaviour on PHP <7, currently, it's not defined by test

Reply:
Yes it is. Take a look at the `testFix` function: if you are running tests with a PHP_VERSION lower than the third param, the second param becomes both the expected and the input values.